macros/db3-check.m4

```diff
@@ -86,7 +86,7 @@ AC_DEFUN([AC_NETATALK_PATH_BDB],[
 if test "x$bdb_required" = "xyes"; then
     trybdbdir=""
     dobdbsearch=yes
-    bdb_search_dirs="/usr/local /usr"
+    bdb_search_dirs="/usr/local /usr /usr/local/opt/berkeley-db /opt/homebrew/opt/berkeley-db"
```

Comment:
Rather than hard coding these obscure paths, would it be possible to ask macOS users to pass the `--with-bdb=/path/to/bdb/installation/` configure parameter instead?

Reply:
It certainly would be possible but these paths are not obscure if you use a Mac as your primary machine for compilation of open source software.

Reply:
The location is obscure in the sense that it's whatever the homebrew team decided to use at one time, and might be subject to change in the future. Another package manager might become popular and take over, like what happened when macports lost popularity.

I think the --with-bdb override exists just for this kind of usecase -- an unofficial package manager and non-standard library locations.

Of course I do see the convenience factor for macos users, but I wanted to discuss the sustainable way to manage this macro. It's a bit of a slippery slope to try to support every non standard location...

Reply:
Case in point: OmniOS (an OpenSolaris descendent) installs the Berkeley libs in /opt/ooce so configuring netatalk with `--with-bdb=/opt/ooce` becomes mandatory.

Reply:
Fair do's, I have removed the SSL and db3 macro mods from the changes. By the way, OpenSSL3 breaks DHX authentication when logging in from classic Mac OS clients because it enforces a minimum DH modulus size of 512 bits. DHX (otherwise known as DHCAST128) uses a modulus size of 128 bits so you have to compile against OpenSSL 1.1 or libreSSL to login successfully.

Reply:
I have also updated install.xml to prompt users to specify a non standard SSL install location in the same way as the manual already does for BDB

Reply:
Looking good, thanks for making those changes.

Well noted on the breaking change in openssl3; TIL.

FWIW I have the habit of disabling DHX and using clrtxt for auth with classic Mac OS clients, DHX2 with macOS clients, since DHX was buggy on ARM for the longest time (works now I think.)

AFAICT the upcoming Debian Bookworm has dropped openssl1.1 altogether now so it's a bridge that we have to cross sooner or later...

Reply:
LibreSSL is a good alternative and is likely to maintain support for older protocols for longer.

Reply:
Good point... netatalk3 already supports libressl I think.